### ISSUE TITLE
clojure.java.browse/browse-url not working in NixOS

### ISSUE THREAD

**Issue by thiagokokada** [babashka/babashka#845]
**version**

babashka v0.4.1

**platform**

NixOS

**problem**

In Babashka `clojure.java.browse/browse-url` doesn't work:

```
Babashka v0.4.1 REPL.
Use :repl/quit or :repl/exit to quit the REPL.
Clojure rocks, Bash reaches.

user=> (require '[clojure.java.browse :as b])
nil
user=> (b/browse-url "https://www.google.com")
java.io.IOException: Cannot run program "/usr/bin/xdg-open": error=2, No such file or directory [at <repl>:2:1]2:1]
```

It works in Clojure though:

```
Clojure 1.10.1
user=> (require '[clojure.java.browse :as b])
nil
user=> (b/browse-url "https://www.google.com")
true
```

In NixOS, `xdg-open` is not available in `/usr/bin/xdg-open`, it is in `/run/current-system/sw/bin/xdg-open`. Clojure does the correct thing and tries to auto-detect the binary location: https://github.com/clojure/clojure/blob/ef00c7cffad10e6104d333f7f71a29163b06cd0a/src/clj/clojure/java/browse.clj#L21-L27.

I don't know why Babashka behavior is different here, but the correct behavior would also be to try to auto-detect the location using `which` (or even `babashka.fs/which`).

**repro**

On any minimalistic Linux distro (like Alpine), you can try to put the `xdg-open` binary in some other place in the path and call `(clojure.java.browse/browse-url "https://www.google.com")`.

**expected behavior**

The same one as Clojure.

**Comment by thiagokokada** [comment]
Seeing the implementation of `babashka.impl.clojure.java.browse`, it is clear what is the problem: https://github.com/babashka/babashka/blob/master/src/babashka/impl/clojure/java/browse.clj#L20

I can provide a PR using `babashka.fs/which` to fix this issue. WDYT @borkdude ?

**Comment by thiagokokada** [comment]
Supporting `*open-url-script*` (https://github.com/clojure/clojure/blob/ef00c7cffad10e6104d333f7f71a29163b06cd0a/src/clj/clojure/java/browse.clj#L40) also seems a good idea, since some people in Linux may have a custom script that is used in place of `xdg-open`.

**Comment by thiagokokada** [comment]
Closed by #846.
